bookserver/crud.py

```diff
@@ -153,7 +153,10 @@ async def create_user(user: AuthUserValidator) -> AuthUserValidator:
     the password then add this user to the database.
     """
     if await fetch_user(user.username):
-        raise ValidationError(["user already exists"], AuthUserValidator)
+        raise HTTPException(
+            status_code=422, detail=[{"loc": ["username"], "msg": "duplicate user"}]
```

Comment:
This is a nice direct way to do this.

It does not seem like pydantic.ValidationErrors are meant to be raised by mere mortals.  This results in the same response to the browser without needing to install any additional middleware.

We could probably improve on this with a utility function for adding the detail.

Reply:
Sigh -- I'm not surprised. The simple solution you have here is certainly the best approach, then (with the suggested utility function.)